metadata-ingestion/src/datahub/ingestion/source/hex/mapper.py

```diff
@@ -0,0 +1,330 @@
+import logging
+from datetime import datetime
+from typing import Iterable, List, Optional
+
+from datahub._codegen.aspect import (
+    _Aspect,  # TODO: is there a better import than this one?
+)
+from datahub.emitter.mce_builder import (
+    make_container_urn,
+    make_dashboard_urn,
+    make_data_platform_urn,
+    make_dataplatform_instance_urn,
+    make_tag_urn,
+    make_ts_millis,
+    make_user_urn,
+)
+from datahub.emitter.mcp import MetadataChangeProposalWrapper
+from datahub.emitter.mcp_builder import DatahubKey
+from datahub.ingestion.api.incremental_lineage_helper import (
+    convert_dashboard_info_to_patch,
+)
+from datahub.ingestion.api.workunit import MetadataWorkUnit
+from datahub.ingestion.source.common.subtypes import BIAssetSubTypes
+from datahub.ingestion.source.hex.constants import (
+    HEX_API_BASE_URL_DEFAULT,
+    HEX_PLATFORM_NAME,
+)
+from datahub.ingestion.source.hex.model import (
+    Analytics,
+    Category,
+    Collection,
+    Component,
+    Owner,
+    Project,
+    Status,
+)
+from datahub.metadata.com.linkedin.pegasus2avro.common import (
+    AuditStampClass,
+    ChangeAuditStampsClass,
+    OwnershipType,
+)
+from datahub.metadata.schema_classes import (
+    ContainerClass,
+    ContainerPropertiesClass,
+    DashboardInfoClass,
+    DashboardUsageStatisticsClass,
+    DataPlatformInstanceClass,
+    GlobalTagsClass,
+    OwnerClass,
+    OwnershipClass,
+    SubTypesClass,
+    TagAssociationClass,
+)
+from datahub.metadata.urns import ContainerUrn, CorpUserUrn, DashboardUrn, Urn
+
+logger = logging.getLogger(__name__)
+
+
+class WorkspaceKey(DatahubKey):
```

Comment:
if you inherit from `ContainerKey`, you automatically get platform/instance fields and a `as_urn` / `as_urn_typed` method

Reply:
We also get optional `env` 🤔 

We mentioned that `platform_instance` wouldn't be needed for this connector, however I added it to the source config and so here in the key as well as other asset URNs. Basically:
- we left to the user the decision or setting platform instance or not; in the unlikely but possible case of users having multiple single-tenant hex contracts, platform instance may be useful
- also, this is aligned with some other BI ingestors, which also define platform instance as optional (eg powerbi, tableau, ...)

Same rationale would apply to `env`, right?

Reply:
yes - so we can leave platform_instance / env as None when creating the `WorkspaceKey`

Reply:
addressed in commit a1243af203c2a3b7b229555fa8082f2b700b0131